scripts/grace_sweep_report.py

```
@@ -0,0 +1,202 @@
+            out.append(f"  paired at all : 0/{len(legs)} -- no companion ever filled")
+        out.append("")
+
+        out.append("EVERY GRACE VALUE, FROM THE ONE RUN")
+        out.append("  rescued = companion filled inside the horizon, so the pair merges")
+        out.append(f"  the {POLL_INTERVAL_SEC:.0f}s row is the floor: the loop cannot "
+                   f"exit sooner than its own poll")
+        out.append("  exit    = the rest, sold at the bid as it stood at the horizon")
+        out.append(f"  {'grace':>7}{'rescued':>10}{'rate':>9}"
+                   f"{'merge $':>10}{'exit $':>10}{'net $':>10}")
+        for h in HORIZONS:
+            rescued = [l for l in legs if l["gap"] is not None and l["gap"] <= h]
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)